lib/src/common/disposable.dart

```diff
@@ -21,25 +21,45 @@ import 'package:meta/meta.dart';
 import 'package:w_common/src/common/disposable_manager.dart';
 import 'package:w_common/src/common/managed_stream_subscription.dart';
 
-// ignore: one_member_abstracts
-abstract class _Disposable {
-  Future<Null> dispose();
-}
-
-class InternalDisposable implements _Disposable {
+class InternalDisposable implements SimpleDisposable {
   Disposer _disposer;
+  final Completer<Null> _didDispose = new Completer<Null>();
+  bool _isDisposing = false;
 
   InternalDisposable(this._disposer);
 
   @override
   Future<Null> dispose() {
-    var disposeFuture = _disposer != null ? _disposer() : null;
-    _disposer = null;
-    if (disposeFuture == null) {
-      return new Future.value();
+    if (isDisposed) {
+      return null;
```

Comment:
What's wrong with awaiting null :trollface: ? 

This was done because that's what disposable does https://github.com/Workiva/w_common/blame/master/lib/src/common/disposable.dart#L280. @georgelesica-wf any particular reason? 

I originally wanted to just hand back a full `Disposable` under a limited interface, but that could screw with logging :( so I replicated the behavior for `InternalDisposable`.